eclair-core/src/main/scala/fr/acinq/eclair/db/Databases.scala

```diff
@@ -109,127 +166,52 @@ object Databases extends Logging {
-                   username: Option[String], password: Option[String],
-                   poolProperties: Map[String, Long],
-                   instanceId: UUID,
-                   databaseLeaseInterval: FiniteDuration,
-                   lockExceptionHandler: LockExceptionHandler = { _ => () },
-                   lockType: LockType = LockType.NONE, datadir: File): Databases with ExclusiveLock = {
-    val url = s"jdbc:postgresql://${host}:${port}/${database}"
-
-    checkIfDatabaseUrlIsUnchanged(url, datadir)
-
-    implicit val lock: DatabaseLock = lockType match {
-      case LockType.NONE => NoLock
-      case LockType.LEASE => LeaseLock(instanceId, databaseLeaseInterval, lockExceptionHandler)
-      case _ => throw new RuntimeException(s"Unknown postgres lock type: `$lockType`")
-    }
-
-    import com.zaxxer.hikari.{HikariConfig, HikariDataSource}
-
-    val config = new HikariConfig()
-    config.setJdbcUrl(url)
-    username.foreach(config.setUsername)
-    password.foreach(config.setPassword)
-    poolProperties.get("max-size").foreach(x => config.setMaximumPoolSize(x.toInt))
-    poolProperties.get("connection-timeout").foreach(config.setConnectionTimeout)
-    poolProperties.get("idle-timeout").foreach(config.setIdleTimeout)
-    poolProperties.get("max-life-time").foreach(config.setMaxLifetime)
-
-    implicit val ds: DataSource = new HikariDataSource(config)
-
-    val databases = new Databases with ExclusiveLock {
-      override val network = new PgNetworkDb
-      override val audit = new PgAuditDb
-      override val channels = new PgChannelsDb
-      override val peers = new PgPeersDb
-      override val payments = new PgPaymentsDb
-      override val pendingRelay = new PgPendingRelayDb
-      override def obtainExclusiveLock(): Unit = lock.obtainExclusiveLock
-    }
-    databases.obtainExclusiveLock()
-    databases
-  }
-
-  def sqliteDatabaseByConnections(auditJdbc: Connection, networkJdbc: Connection, eclairJdbc: Connection): Databases = new Databases with FileBackup {
-    override val network = new SqliteNetworkDb(networkJdbc)
-    override val audit = new SqliteAuditDb(auditJdbc)
-    override val channels = new SqliteChannelsDb(eclairJdbc)
-    override val peers = new SqlitePeersDb(eclairJdbc)
-    override val payments = new SqlitePaymentsDb(eclairJdbc)
-    override val pendingRelay = new SqlitePendingRelayDb(eclairJdbc)
-    override def backup(backupFile: File): Unit = {
-
-      SqliteUtils.using(eclairJdbc.createStatement()) {
-        statement => {
-          statement.executeUpdate(s"backup to ${backupFile.getAbsolutePath}")
-        }
-      }
-
     }
   }
 
-  def setupPgDatabases(dbConfig: Config, instanceId: UUID, datadir: File, lockExceptionHandler: LockExceptionHandler): Databases with ExclusiveLock = {
+  def postgres(dbConfig: Config, instanceId: UUID, dbdir: File, lockExceptionHandler: LockFailureHandler = LockFailureHandler.logAndStop)(implicit system: ActorSystem): PostgresDatabases = {
```

Comment:
Why do you need to pull in an implicit ActorSystem?

Reply:
The locking mechanism needs it to schedule the periodic lease refresh.

NB: What we really need is a `Scheduler` and an `ExecutionContext`. Actually it could make sense to create a private independent `ExecutionContext` for that purpose. And if we go that way, since we would have a dedicated thread available, we may as well implement the scheduling part ourselves and remove the implicits altogether. What do you think?